lib/filewatch/watch.rb

```diff
@@ -118,13 +118,21 @@ def each(&block)
       synchronized do
         return if @files.empty?
 
+        quitting = false
         file_deleteable = []
         # creates this array just once
         watched_files = @files.values
 
+        quit_proc = lambda do
```

Comment:
I think this name is a bit misleading, this is more a cleanup task that only runs if quitting is true, and also due to tap's return behaviour, also returns quitting.

I don't have a good alternative other than to make this cleanup task an actual method and be more verbose by doing the 2 steps: `cleanup if quitting; quitting`, or moving "cleanup" to the ensure block of this method? assuming it will always run when leaving this method?

Anyway, it's up to you to change anything. I'm finding the code a bit hard to read, but it's correct.